libs/core/tests/unit_tests/test_tools.py

```diff
@@ -959,6 +959,19 @@ def foo(bar: str, baz: int) -> str:
         "required": ["bar", "baz"],
     }
 
+    foo2 = tool(foo, parse_docstring=True)
```

Comment:
Could you add a test that shows that this throws an error it the doc-string does not follow the correct format?

Reply:
done